streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -184,8 +184,7 @@ class CheckpointWriter(
   val executor = Executors.newFixedThreadPool(1)
   val compressionCodec = CompressionCodec.createCodec(conf)
   private var stopped = false
-  private var _fs: FileSystem = _
-
+  @volatile private[this] var fs: FileSystem = null
```

Comment:
minor: `fs` and `latestCheckpointTime` don't need to be `volatile` as we use `Executors.newFixedThreadPool(1)` here. We should also give the thread a name for better debugging.